ledger/src/blockstore.rs

```diff
@@ -1904,35 +1902,33 @@ impl Blockstore {
             return true;
         };
 
-        let Some(prev_merkle_root_meta) = merkle_root_metas
+        let Some(prev_erasure_meta) = erasure_metas
             .get(&prev_erasure_set)
             .map(WorkingEntry::as_ref)
             .map(Cow::Borrowed)
-            .or_else(|| {
-                self.merkle_root_meta(prev_erasure_set)
-                    .unwrap()
-                    .map(Cow::Owned)
-            })
+            .or_else(|| self.erasure_meta(prev_erasure_set).unwrap().map(Cow::Owned))
         else {
-            warn!(
-                "The merkle root meta for the previous erasure set {prev_erasure_set:?} does not exist.
-                This should only happen if you have recently upgraded from a version < v1.18.13.
-                Skipping the backwards chained merkle root for {erasure_set:?}"
+            error!(
+                "The erasure meta for the previous erasure set {prev_erasure_set:?} does not exist.
+                 This should only happen in extreme cases where blockstore cleanup has caught up to the root.
+                 Skipping the backwards chained merkle root consistency check for {erasure_set:?}"
             );
             return true;
         };
         let prev_shred_id = ShredId::new(
             slot,
-            prev_merkle_root_meta.first_received_shred_index(),
-            prev_merkle_root_meta.first_received_shred_type(),
+            prev_erasure_meta
+                .first_received_coding_shred_index()
+                .expect("First received coding index must fit in u32"),
+            ShredType::Code,
         );
         let Some(prev_shred) =
             Self::get_shred_from_just_inserted_or_db(self, just_inserted_shreds, prev_shred_id)
                 .map(Cow::into_owned)
         else {
-            error!(
-                "Shred {prev_shred_id:?} indicated by merkle root meta {prev_merkle_root_meta:?} is missing from blockstore.
-                 This should only happen in extreme cases where blockstore cleanup has caught up to the root.
+            warn!(
```

Comment:
We enter this if either:
* Blockstore cleanup has caught up to root and `prev_shred` has been cleaned up
* `prev_erasure_meta` was inserted with an older version and `prev_erasure_meta.first_received_coding_shred_index()` points to shred 0 which we do not have.